### ISSUE TITLE
on fresh install - no module named api (for remote runner)

### ISSUE THREAD

**Issue by drahmany** [avocado-framework/avocado#2628]
Failed to load plugin from module "avocado_runner_remote": ImportError('No module named api',)

suspicious line during install:

Requirement already satisfied: cffi>=1.7; platform_python_implementation != "PyPy" in ./lib/python2.7/site-packages (from cryptography>=1.1->fabric->avocado-framework-plugin-runner-remote==54.0) (1.11.5)

happened with both 54 and 61.

**Comment by drahmany** [comment]
[@#[](url)](url)

**Comment by drahmany** [comment]
turns out fabric may have been updated so fabric.api wasn't working. I am using 1.14.0 and it seems to work fine again. Is there a better way to limit the dependencies? This is the 4th or 5th time we have had to force versions to keep avocado working.

**Comment by ldoktor** [comment]
Hello guys,

I'm sorry to hear that. Requirements are always tricky as other project might just release a new version unnoticed and we don't want to restrict the newer versions. I looked at it and the problem is that Fabric introduced 2.x.y release, which is a complete rework. PR to limit us to 1.x.y is here https://github.com/avocado-framework/avocado/pull/2645

We can try adding some selftests to avoid such issues, but without a real ssh in travis we'll never be sure.

**Comment by ldoktor** [comment]
Actually we do have selftests that fail with the latest fabric, so it should have been caught by travis. It was not because of cliff broken dep to-be-fixed in https://github.com/avocado-framework/avocado/pull/2643 which prevented travis py2 checks. Anyway trying `make check` on my machine with fabric2:

```
Running 'inspekt lint --exclude=.git --enable W0101,W0102,W0404,W0611,W0612,W0622'
Pylint disabled: W,R,C,E1002,E1101,E1103,E1120,F0401,I0011
Pylint enabled : W0101,W0102,W0404,W0611,W0612,W0622
************* Module avocado_runner_remote
E0611: 24,0: : No name 'api' in module 'fabric'
E0611: 25,0: : No name 'network' in module 'fabric'
E0611: 26,0: : No name 'operations' in module 'fabric'
E0611: 27,0: : No name 'tasks' in module 'fabric'
E0611: 28,0: : No name 'context_managers' in module 'fabric'
E0611: 29,0: : No name 'CommandTimeout' in module 'fabric.exceptions'
Syntax check FAIL
lint FAILED
```

Anyway remember with `requirements.txt` you are always tracing the latest and occasionally unstable versions so in production I'd recommend using custom `requirements` file or ideally distribution packages, which follow the upstram with small delay. Fabric 2.0.0 was released 20 days ago, now it's on version 2.1.3 because of many bug fixes.

**Comment by ldoktor** [comment]
the requirements fix had been merged.
